jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -136,12 +137,32 @@ public static Booster train(
           IEvaluation eval,
           int earlyStoppingRound,
           Booster booster) throws XGBoostError {
+    if(eval != null) {
+      return trainWithMultipleEvals(dtrain, params, round, watches, metrics, obj,
+                                    new IEvaluation[]{eval}, earlyStoppingRound, booster);
+    } else {
+      return trainWithMultipleEvals(dtrain, params, round, watches, metrics, obj,
+                                    null, earlyStoppingRound, booster);
+    }
+  }
+
+  public static BoosterResults trainWithResults(
+      DMatrix dtrain,
+      Map<String, Object> params,
+      int round,
+      Map<String, DMatrix> watches,
+      float[][] metrics,
+      IObjective obj,
+      IEvaluation[] evals,
+      int earlyStoppingRound,
+      Booster booster) throws XGBoostError {
```

Comment:
it looks like you have empty evals and null both indicate no customized evaluation function

to make the logic clear, we'd make only one of them to be a valid input and the other triggers an exception

Reply:
If there are no evaluation functions, I chose to allow null and throw error on empty array of evaluation fucntions.